cmd/libsnap-confine-private/cgroup-pids-support.c

```diff
@@ -0,0 +1,57 @@
+// For AT_EMPTY_PATH and O_PATH
+#define _GNU_SOURCE
+
+#include "cgroup-pids-support.h"
+
+#include <errno.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "cleanup-funcs.h"
+#include "string-utils.h"
+#include "utils.h"
+
+static const char *pids_cgroup_dir = "/sys/fs/cgroup/pids";
+
+void sc_cgroup_pids_join(const char *snap_security_tag, pid_t pid) {
+    // Open the pids cgroup directory.
```

Comment:
We have very similar code in sc_cgroup_freezer_join at leat the fist ~30 lines or so look quite similar - is there a chance that we can consolidate these two functions?

Reply:
Once we get a third version, yes, before that I'd rather not.

Reply:
I respectfully disagree, it seems something like  http://paste.ubuntu.com/p/SSg3qf5xtj/ is all that is needed to unify the two helpers and I don't think we want duplication like this in something we merge into master even (or maybe especially) if we know this will change anyway. It means the reviewers will have to review this carefully (this is suid root code afterall) and when this is changed again do same same. Or am I missing something here?

Reply:
https://en.wikipedia.org/wiki/Rule_of_three

Reply:
I agree with @mvo5 here,  we don't need 3, it's the nature of cgroups to have special "tasks" files to which you write pid to add to them. Is the freezer cgroup "tasks" different from the "tasks" here?  apart being in different places. Notice that writing this again completely you are asking people to re-review the same code essentially again.

Reply:
I'll refactor this and post it back.

Reply:
Done now